From 51084b3270d17f0d75893329737e5baf48a040bc Mon Sep 17 00:00:00 2001
From: rubin <rubin@126.com>
Date: Mon, 3 Jun 2024 02:04:39 +0000
Subject: [PATCH] general-modify-edp-panel

---
 drivers/gpu/drm/panel/panel-edp.c | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/drivers/gpu/drm/panel/panel-edp.c b/drivers/gpu/drm/panel/panel-edp.c
index 70feee787611..bff2b28ba6f9 100644
--- a/drivers/gpu/drm/panel/panel-edp.c
+++ b/drivers/gpu/drm/panel/panel-edp.c
@@ -1722,6 +1722,28 @@ static const struct panel_desc starry_kr122ea0sra = {
 		.unprepare = 10 + 500,
 	},
 };
+static const struct drm_display_mode xiyun_unknownpanel_mode = {
+	.clock = 97760,
+	.hdisplay = 1600,
+	.hsync_start = 1600 + 48,
+	.hsync_end = 1600 + 48 + 32,
+	.htotal = 1600 + 48 + 32 + 80,
+	.vdisplay = 900,
+	.vsync_start = 900 + 3,
+	.vsync_end = 900 + 3 + 5,
+	.vtotal = 900 + 3 + 5 + 18,
+	.flags = DRM_MODE_FLAG_NVSYNC | DRM_MODE_FLAG_NHSYNC ,
+};
+
+static const struct panel_desc xiyun_unknownpanel = {
+	.modes = &xiyun_unknownpanel_mode,
+	.num_modes = 1,
+	.bpc = 8,
+	.size = {
+		.width = 260,
+		.height = 150,
+	},
+};
 
 static const struct of_device_id platform_of_match[] = {
 	{
@@ -1811,6 +1833,9 @@ static const struct of_device_id platform_of_match[] = {
 	}, {
 		.compatible = "starry,kr122ea0sra",
 		.data = &starry_kr122ea0sra,
+	}, {
+		.compatible = "xiyun,unknown-panel",
+		.data = &xiyun_unknownpanel,
 	}, {
 		/* sentinel */
 	}
-- 
2.30.2

